engine/src/test/java/io/zeebe/engine/processing/deployment/model/validation/SuccessfulDeploymentTest.java

```diff
@@ -23,17 +41,70 @@
   public final RecordingExporterTestWatcher recordingExporterTestWatcher =
       new RecordingExporterTestWatcher();
 
+  @Parameter(0)
+  public String description;
+
+  @Parameter(1)
+  public Function<DeploymentClient, Record<DeploymentRecordValue>> performDeployment;
+
+  @Parameters(name = "{0}")
+  public static Collection<Object[]> data() {
+    return Arrays.asList(
+        new Object[] {"orphan error definition", deploy("/processes/orphan-error-definition.bpmn")},
+        new Object[] {
+          "none start event",
+          deploy(Bpmn.createExecutableProcess("process").startEvent().endEvent().done())
+        },
+        new Object[] {
+          "timer start event",
+          deploy(
+              Bpmn.createExecutableProcess("process")
+                  .startEvent()
+                  .timerWithCycle("R/PT10S")
+                  .endEvent()
+                  .done())
+        },
+        new Object[] {
+          "message start event",
+          deploy(
+              Bpmn.createExecutableProcess("process")
+                  .startEvent()
+                  .message("start")
+                  .endEvent()
+                  .done())
+        });
+  }
+
   @Test
-  public void shouldDeployProcessWithOrphanErrorDefinition() {
+  public void shouldWriteDeploymentCreatedEvent() {
     // when
-    final var deployment =
-        engine
-            .deployment()
-            .withXmlClasspathResource("/processes/orphan-error-definition.bpmn")
-            .deploy();
+    final var deployment = performDeployment.apply(engine.deployment());
 
     // then
     assertThat(deployment.getIntent()).isEqualTo(DeploymentIntent.CREATED);
     assertThat(deployment.getValue().getDeployedProcesses()).hasSize(1);
   }
+
+  @Test
+  public void shouldSendResponse() {
```

Comment:
Do we want also one for rejection response?

Reply:
In general, it is a good idea to have a test in the engine. However, we have one in the QA module 
`io.zeebe.broker.it.client.command.CreateDeploymentTest#shouldRejectDeployIfProcessIsInvalid`.